tests/torchtune/utils/test_logging.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+from torchtune.data._utils import deprecated
+
+
+def test_deprecated():
+    @deprecated(msg="Please use `TotallyAwesomeClass` instead.")
+    class DummyClass:
+        pass
+
+    with pytest.warns(
+        FutureWarning,
+        match="DummyClass is deprecated and will be removed in future versions. Please use `TotallyAwesomeClass` instead.",
```

Comment:
this is a nit but it's a bit awkward to me that we split the log message across the utility and the arg we pass. Like your passing msg="Please use `TotallyAwesomeClass` instead" assumes that you know exactly what the first half of the message is, which is somewhat annoying to go check every time

Reply:
Yeah, I could just pass in the full message everytime but that gets tedious. I supposed I could make it a required argument and do that for simplicity.